app/services/GetMessagesService.scala

```diff
@@ -30,30 +30,50 @@ import scala.concurrent.{ExecutionContext, Future}
 class GetMessagesService @Inject()(
                                     messagesConnector: GetMessagesConnector,
                                     markMessageAsReadConnector: MarkMessageAsReadConnector,
+                                    getSubmissionFailureMessageConnector: GetSubmissionFailureMessageConnector,
                                     messageInboxRepository: MessageInboxRepository
                                   )(implicit ec: ExecutionContext) {
 
   def getMessages(ern: String, search: Option[MessagesSearchOptions])(implicit hc: HeaderCarrier): Future[GetMessagesResponse] = {
     messagesConnector.getMessages(ern, search).map {
       case Right(messages) =>
-        messages.messages.foreach { message =>
-          messageInboxRepository.set(MessageCache(ern, message))
-        }
+        messages.messages.foreach(message => messageInboxRepository.set(MessageCache(ern, message, errorMessage = None)))
         messages
       case _ => throw MessagesException(s"Error occurred when fetching messages for trader $ern")
     }
   }
 
   def getMessage(ern: String, uniqueMessageIdentifier: Long)(implicit hc: HeaderCarrier): Future[Option[MessageCache]] = {
 
-    messageInboxRepository.get(ern, uniqueMessageIdentifier).map {
-      case Some(message) =>
-        markMessageAsReadConnector.markMessageAsRead(ern, uniqueMessageIdentifier)
-        Some(message)
+    messageInboxRepository.get(ern, uniqueMessageIdentifier).flatMap {
+      case Some(cachedMessage) =>
+        retrieveSubmissionFailureMessageIfMessageIs704(cachedMessage).map {
+          cachedMessageWithPossible704 =>
+            markMessageAsReadConnector.markMessageAsRead(ern, uniqueMessageIdentifier)
+            Some(cachedMessageWithPossible704)
+        }
       case _ =>
-        None
+        Future(None)
     }
 
   }
 
+  private[services] def retrieveSubmissionFailureMessageIfMessageIs704(cachedMessage: MessageCache)(implicit hc: HeaderCarrier): Future[MessageCache] = {
+    (cachedMessage.message.isAnErrorMessage, cachedMessage.errorMessage) match {
+      case (false, _) | (true, Some(_)) => Future(cachedMessage)
+      case (true, None) =>
+        getSubmissionFailureMessageConnector.getSubmissionFailureMessage(cachedMessage.ern, cachedMessage.message.uniqueMessageIdentifier).map {
+        _.fold(
```

Comment:
would probably prefer case Left/Right here over the fold as it's a bit clearer you're working with an Either